doc/source/ray-core/gpu-objects.rst

```diff
@@ -0,0 +1,212 @@
+GPU objects implement the familiar Ray :class:`ObjectRef <ray.ObjectRef>` API but they also:
+
+- Keep data in GPU memory until a transfer is needed
+- Avoid expensive serialization and copies to and from CPU memory
+- Use efficient data transports like collective communication libraries or RDMA (via `NVIDIA's NIXL <https://github.com/ai-dynamo/nixl>`__) to transfer data between GPUs
+
+.. note::
+   The GPU objects feature is currently in **alpha**. Not all Ray Core APIs are supported yet. Future releases may introduce breaking API changes.
+
+Getting started
+---------------
+
+GPU objects currently support ``torch.Tensor`` objects created by Ray actor tasks.
+This walkthrough will show how to create and use GPU objects using different *tensor transports*, i.e. the mechanism used to transfer the tensor between actors.
+Currently, we support:
+
+1. `Gloo <https://github.com/pytorch/gloo>`__: A collective communication library for PyTorch and CPUs.
+2. `NVIDIA NCCL <https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/index.html>`__: A collective communication library for NVIDIA GPUs.
+3. `NVIDIA NIXL <https://github.com/ai-dynamo/nixl>`__ (backed by `UCX <https://github.com/openucx/ucx>`__): A library for accelerating point-to-point transfers via RDMA, especially between various types of memory and NVIDIA GPUs.
+
+For ease of following along, we'll start with the `Gloo <https://github.com/pytorch/gloo>`__ transport, which can be used without any physical GPUs.
+
+
+Installation
+------------
+
+.. note::
+    Under construction.
+
+.. _gpu-objects-gloo:
+Usage with gloo (CPUs only)
+---------------------------
+
+Installation
+************
+
+.. note::
+    Under construction.
+
+Walkthrough
+***********
+
+To get started, define an actor class and a task that returns a ``torch.Tensor``:
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __normal_example_start__
+   :end-before: __normal_example_end__
+
+As written, when the ``torch.Tensor`` is returned, it will be copied into Ray's CPU-based object store.
+For CPU-based tensors, this can require an expensive step to copy and serialize the object, while GPU-based tensors additionally require a copy to and from CPU memory.
+
+To enable GPU objects, use the ``tensor_transport`` option in the :func:`@ray.method <ray.method>` decorator.
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __gloo_example_start__
+   :end-before: __gloo_example_end__
+
+This decorator can be added to any actor tasks that return a ``torch.Tensor``, or that return ``torch.Tensors`` nested inside other Python objects.
+Adding this decorator will change Ray's behavior in the following ways:
+
+1. When returning the tensor, Ray will store a *reference* to the tensor instead of copying it to CPU memory. If multiple tensors are found, then Ray will store a reference for each tensor.
```

Comment:
```suggestion
1. When returning the tensor, Ray will store a *reference* to the tensor instead of copying it to CPU memory. If multiple tensors are contained in the return object, then Ray will store a reference for each tensor.
```

Reply:
consider not mentioning multiple tensors here and leaving that to a standalone section